core/src/test/java/com/novoda/noplayer/internal/exoplayer/AdvertPlaybackStateTest.java

```diff
@@ -47,19 +50,88 @@
         assertThat(adPlaybackState.adGroupTimesUs).containsSequence(ONE_SECOND_IN_MICROS, TWO_SECONDS_IN_MICROS, THREE_SECONDS_IN_MICROS);
-        assertThatGroupContains(adPlaybackState.adGroups[0], 1, new long[]{ONE_SECOND_IN_MICROS}, new Uri[]{FIRST_ADVERT.uri()});
-        assertThatGroupContains(adPlaybackState.adGroups[1], 2, new long[]{ONE_SECOND_IN_MICROS, TWO_SECONDS_IN_MICROS}, new Uri[]{FIRST_ADVERT.uri(), SECOND_ADVERT.uri()});
-        assertThatGroupContains(adPlaybackState.adGroups[2], 3, new long[]{ONE_SECOND_IN_MICROS, TWO_SECONDS_IN_MICROS, THREE_SECONDS_IN_MICROS}, new Uri[]{FIRST_ADVERT.uri(), SECOND_ADVERT.uri(), THIRD_ADVERT.uri()});
+        assertThat(adPlaybackState.adGroups).containsExactly(FIRST_AD_GROUP, SECOND_AD_GROUP, THIRD_AD_GROUP);
+        assertThat(adPlaybackState.adResumePositionUs).isEqualTo(0L);
+        assertThat(adPlaybackState.contentDurationUs).isEqualTo(C.TIME_UNSET);
+    }
+
+    @Test
+    public void createsEmptyStateWhenAdvertBreaksAreEmpty() {
+        List<AdvertBreak> advertBreaks = Collections.emptyList();
+
+        AdvertPlaybackState advertPlaybackState = AdvertPlaybackState.from(advertBreaks, HALF_SECOND_IN_MILLIS);
+        AdPlaybackState adPlaybackState = advertPlaybackState.adPlaybackState();
+
+        assertThat(adPlaybackState.adGroupCount).isEqualTo(0);
+        assertThat(adPlaybackState.adGroupTimesUs).isEmpty();
+        assertThat(adPlaybackState.adGroups).isEmpty();
+        assertThat(adPlaybackState.adResumePositionUs).isEqualTo(0L);
+        assertThat(adPlaybackState.contentDurationUs).isEqualTo(C.TIME_UNSET);
+    }
+
+    @Test
+    public void createsAdvertPlaybackStateWithResumePositionInMicroseconds() {
+        List<AdvertBreak> advertBreaks = Collections.singletonList(FIRST_ADVERT_BREAK);
+
+        AdvertPlaybackState advertPlaybackState = AdvertPlaybackState.from(advertBreaks, HALF_SECOND_IN_MILLIS);
+        AdPlaybackState adPlaybackState = advertPlaybackState.adPlaybackState();
+
+        assertThat(adPlaybackState.adResumePositionUs).isEqualTo(HALF_SECOND_IN_MICROS);
+    }
+
+    @Test
+    public void marksAdvertsInAdvertBreakAsPlayedWhenResumePositionIsMoreThanEachAdvertDuration() {
+        List<AdvertBreak> advertBreaks = Collections.singletonList(THIRD_ADVERT_BREAK);
+        AdPlaybackState.AdGroup expectedAdGroup = thirdAdGroupFixture()
+                .withPlayedStateAt(0)
+                .withPlayedStateAt(1)
+                .build();
+        long resumePosition = FIRST_ADVERT.durationInMillis() + SECOND_ADVERT.durationInMillis() + HALF_SECOND_IN_MILLIS;
+
+        AdvertPlaybackState advertPlaybackState = AdvertPlaybackState.from(advertBreaks, resumePosition);
+        AdPlaybackState adPlaybackState = advertPlaybackState.adPlaybackState();
+
+        assertThat(adPlaybackState.adGroups).containsExactly(expectedAdGroup);
+        assertThat(adPlaybackState.adResumePositionUs).isEqualTo(HALF_SECOND_IN_MICROS);
```

Comment:
We should be able to assert here that `adPlaybackState` `STATE` is `AD_STATE_PLAYED`; right? 

And that will honor even more the name of the test

Reply:
I guess that's part of the weirdness of the `AdPlaybackState`. At the top of the test lines `99` and `100` we create the expected group with the played adverts at the expected positions. But then the `adResumePositionUs` is not per group but it's global in the whole `AdPlaybackState` so that's why we only assert it here.